persistence/src/main/java/de/cyface/persistence/DatabaseHelper.java

```diff
@@ -89,44 +90,51 @@ public void onCreate(final SQLiteDatabase db) {
 
     /**
      * The onUpgrade method is called when the app is upgraded and the DATABASE_VERSION changed.
-     * The incremental database upgrades are executed to reach the current version.
+     * <p>
+     * This method is not called incrementally by the system which is why this method implements this.
+     * <p>
+     * This method is automatically executed in a transaction, do not wrap the code in another transaction!
      *
      * @param database the database which shall be upgraded
      * @param oldVersion the database version the app was in before the upgrade
      * @param newVersion the database version of the new, upgraded app which shall be reached
      */
     @Override
     public void onUpgrade(final SQLiteDatabase database, final int oldVersion, final int newVersion) {
-        Log.w(TAG, "Upgrading database from version " + oldVersion + " to " + newVersion);
+        Validate.isTrue(oldVersion == 8 || oldVersion >= 11, "Unsupported versions");
 
-        // Incremental upgrades for the tables which don't exist anymore and, thus, don't have an own class file anymore
-        // noinspection SwitchStatementWithTooFewBranches - because others will follow and it's an easier read
-        switch (oldVersion) {
-            case 8:
-                // This upgrade from 8 to 10 is executed for all SDK versions below 3 (which is v 10).
-                // We don't support an soft-upgrade there but reset the database
-                Log.w(TAG, "Upgrading from version " + oldVersion + " to " + newVersion + ": Resetting database");
+        // Upgrade incrementally to reduce the amount of migration code required
+        for (int fromVersion = oldVersion; fromVersion < newVersion; fromVersion++) {
```

Comment:
All onUpgrade Sections - in here and in the other Tables are now called incrementally, i.e. 8->9, 9->10, etc. which makes it easier to understand.